src/base/bittorrent/customstorage.cpp

```diff
@@ -35,6 +35,191 @@
+        if (!error)
+            m_storageData[storage].files.rename_file(index, name);
+        handler(name, index, error);
+    });
+}
+
+void customDiskIOThread::async_delete_files(lt::storage_index_t storage, lt::remove_flags_t options
+                                            , std::function<void (const lt::storage_error &)> handler)
+{
+    m_nativeDiskIO->async_delete_files(storage, options, handler);
+}
+
+void customDiskIOThread::async_set_file_priority(lt::storage_index_t storage, lt::aux::vector<lt::download_priority_t, lt::file_index_t> priorities
+                                                 , std::function<void (const lt::storage_error &, lt::aux::vector<lt::download_priority_t, lt::file_index_t>)> handler)
+{
+    m_nativeDiskIO->async_set_file_priority(storage, priorities
+                                            , [=](const lt::storage_error &error, lt::aux::vector<lt::download_priority_t, lt::file_index_t> priorities)
+    {
+        m_storageData[storage].filePriorities = priorities;
+        handler(error, priorities);
+    });
+}
+
+void customDiskIOThread::async_clear_piece(lt::storage_index_t storage, lt::piece_index_t index
+                                           , std::function<void (lt::piece_index_t)> handler)
+{
+    m_nativeDiskIO->async_clear_piece(storage, index, handler);
+}
+
+void customDiskIOThread::update_stats_counters(lt::counters &counters) const
+{
+    m_nativeDiskIO->update_stats_counters(counters);
+}
+
+std::vector<lt::open_file_state> customDiskIOThread::get_status(lt::storage_index_t index) const
+{
+    return m_nativeDiskIO->get_status(index);
+}
+
+void customDiskIOThread::abort(bool wait)
+{
+    m_nativeDiskIO->abort(wait);
+}
+
+void customDiskIOThread::submit_jobs()
+{
+    m_nativeDiskIO->submit_jobs();
+}
+
+void customDiskIOThread::settings_updated()
+{
+    m_nativeDiskIO->settings_updated();
+}
+
+void customDiskIOThread::handleCompleteFiles(lt::storage_index_t storage, const QString &savePath)
+{
+    const QDir saveDir {savePath};
+    const StorageData storageData = m_storageData[storage];
+    const lt::file_storage &fileStorage = storageData.files;
+    for (const lt::file_index_t fileIndex : fileStorage.file_range()) {
+        // ignore files that have priority 0
+        if ((storageData.filePriorities.end_index() > fileIndex) && (storageData.filePriorities[fileIndex] == lt::dont_download))
+            continue;
+
+        // ignore pad files
+        if (fileStorage.pad_file_at(fileIndex)) continue;
+
+        const QString filePath = QString::fromStdString(fileStorage.file_path(fileIndex));
+        if (filePath.endsWith(QB_EXT)) {
+            const QString completeFilePath = filePath.left(filePath.size() - QB_EXT.size());
+            QFile completeFile {saveDir.absoluteFilePath(completeFilePath)};
+            if (completeFile.exists()) {
+                QFile incompleteFile {saveDir.absoluteFilePath(filePath)};
+                if (incompleteFile.exists())
```

Comment:
I think you can just call `remove()` without checking it first.

Reply:
Oh, now I noticed this code is copied from another #if branch. Do you intend to make changes over there too?